ntoskrnl/ke/i386/mp.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Architecture source file to hold multiprocessor functions
+ * COPYRIGHT:   Copyright 2021 Justin Miller <justinmiller100@gmail.com>
+ */
+
+/* INCLUDES *****************************************************************/
+
+#include <ntoskrnl.h>
+#define NDEBUG
+#include <debug.h>
+
+/* GLOBALS *******************************************************************/
+
+#define PHYSICAL_ADDRESS  LARGE_INTEGER
+PHYSICAL_ADDRESS HighestPhysicalAddress;
+ULONG ProcessorCount = 0;
+
+struct APInfo
+{
+    KPCR pcr;
+    KTSS tss;
+    ETHREAD thread;
+    KDESCRIPTOR GdtDesc;
+    KDESCRIPTOR IdtDesc;
+} APInfo;
+
+/* FUNCTIONS *****************************************************************/
+VOID
+NTAPI
+KeStartAllProcessors()
```

Comment:
A lot of this function is not architecture dependent. Please split into an architecture independent version, which calls architecture dependent helper function to initialize the processor state.

Reply:
I got a few different ideas for this subject, I'm going to leave this comment open so we can all figure it out.
I have written both versions, where I have done this and where I haven't here is my 2 cents:
- A lot of this function IS architecture dependent, and even the size of for example the TSS is different for x64. 
- I don't have much of an opinion to splitting off the processor state functions but if it's different for every architecture and we are using different implementations of the function per architecture then Is there still a point to splitting it off to another function?
- I feel like separating the implementations is cleaner and easier to work with, it would take a lot of helper functions for selective tasks and/or a lot of preprocessor definitions to get it working as a single implementation. 
- and finally the main reason: This is what my mentor suggested me to do so I PRed this version

Im totally willing to change it if we decide it's a better approach as the file is already written though.

Reply:
Ok so here are my arguments:
1. A lot of this function IS architecture independent
2. Splitting it would result in 1 relatively simple loop + 2 relatively simple flat functions, vs 2 functions with large loop bodies that work partly the same, but partly not and when there is a bug fixed in one, is it gonna be fixed in the other one?
3. No, it doesn't need a lot of helper functions, it only needs one main function and one small helper function per architecture.
4. You should probably address this last, when everything else is addressed.
4. Your mentor better shows up here and explains himself 😛

Reply:
I'd like to postpone the split to later.

1. Currently, we can't check if the functions works properly and I have a feeling it will need more things to be added. When there is a _working_ implementation of AP bootup, we can start refactorings like that. 
2. What's common here? HalStartNextProcessor call, MmCreateKernelStack and KeLoaderBlock initialization? I feel like it's not enough to do the split, currently you can see all actions done in one place, with your approach small functions will be spread across files (so harder to read code)
3. @DarkFire01 Don't put arguments like "This is what my mentor suggested me to do so I PRed this version", that's not constructive. Of you haven't understood my arguments, ask me :)

Reply:
Understood I will switch to the other one!